### PR TITLE
FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] FWI-1618 - [insights-agent|admission] auto-create RBAC rules for OPA target-resources and admission rules

### PR DESCRIPTION
**Pull request by ivanfetch-wt**
**Why This PR?**
This avoids users having to specify RBAC rules to match OPA
target-resources or admission rules.

* RBAC rules are always created for
OPA `defaultTargetResources` and admission `webhookConfig.defaultRules`.
* RBAC rule auto-creation can be disabled for OPA `rules` or admission
`webhookConfig.rules` if users prefer to manage their own RBAC in this
regard.

**Checklist:**

* [x] I have included the name of the chart in the title of this PR in square brackets i.e. `[stable/goldilocks]`.
* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] Any new values have been added to the README for the Chart, or `helm-docs --sort-values-order=file` has been run for the charts that support it.

[Internal Ticket FWI-1618](https://fairwinds.myjetbrains.com/youtrack/issue/FWI-1618)

### CODE REVIEW

File: stable/insights-admission/README.md [hunk 1]
```diff
@@ -51,8 +51,9 @@ rules:
...
+| webhookConfig.rules | list | `[]` | An array of additional rules for the ValidatingWebhookConfiguration. Each requires a set of apiGroups, apiVersions, operations, resources, and a scope. Rules specified here may also be granted to the Insights OPA plugin, see also the insights-al  qgent chart values for opa. |
 | webhookConfig.defaultRules | list | `[{"apiGroups":["apps"],"apiVersions":["v1","v1beta1","v1beta2"],"operations":["CREATE","UPDATE"],"resources":["daemonsets","deployments","statefulsets"],"scope":"Namespaced"},{"apiGroups":["batch"],"apiVersions":["v1","v1beta1"],"operations":["CREATE","UPDATE"],"resources":["jobs","cronjobs"],"scope":"Namespaced"},{"apiGroups":[""],"apiVersions":["v1"],"operations":["CREATE","UPDATE"],"resources":["pods","replicationcontrollers"],"scope":"Namespaced"}]` | An array of rules for commons types for the ValidatingWebhookConfiguration |
+| webhookConfig.rulesAutoRBAC | bool | `true` | AUtomatically add RBAC rules allowing get and list operations for the APIGroups and Resources supplied in rules. This *does not* impact RBAC rules added for `defaultRules`. |
```
Comment (rbren) [inline stable/insights-admission/README.md]:
```suggestion
| webhookConfig.rulesAutoRBAC | bool | `true` | Automatically add RBAC rules allowing get and list operations for the APIGroups and Resources supplied in rules. This *does not* impact RBAC rules added for `defaultRules`. |
```

File: stable/insights-admission/README.md [hunk 2]
```diff
@@ -51,8 +51,9 @@ rules:
 | webhookConfig.matchPolicy | string | `"Exact"` | matchPolicy for the ValidatingWebhookConfiguration |
 | webhookConfig.namespaceSelector | object | `{"matchExpressions":[{"key":"control-plane","operator":"DoesNotExist"}]}` | namespaceSelector for the ValidatingWebhookConfiguration |
 | webhookConfig.objectSelector | object | `{}` | objectSelector for the ValidatingWebhookConfiguration |
-| webhookConfig.rules | list | `[]` | An array of additional rules for the ValidatingWebhookConfiguration. Each requires a set of apiGroups, apiVersions, operations, resources, and a scope. Rules specified here may also be granted to the Insights OPA plugin, see also the insights-agent chart values for opa. |
+| webhookConfig.rules | list | `[]` | An array of additional rules for the ValidatingWebhookConfiguration. Each requires a set of apiGroups, apiVersions, operations, resources, and a scope. Rules specified here may also be granted to the Insights OPA plugin, see also the insights-al  qgent chart values for opa. |
```
Comment (rbren) [inline stable/insights-admission/README.md]:
```suggestion
| webhookConfig.rules | list | `[]` | An array of additional rules for the ValidatingWebhookConfiguration. Each requires a set of apiGroups, apiVersions, operations, resources, and a scope. Rules specified here may also be granted to the Insights OPA plugin, see also the insights-agent chart values for opa. |
```

File: stable/insights-admission/values.yaml [hunk 3]
```diff
@@ -38,7 +38,7 @@ webhookConfig:
   objectSelector: {}
   # webhookConfig.rules -- An array of additional rules for the ValidatingWebhookConfiguration. Each requires a set of apiGroups, apiVersions, operations, resources, and a scope.
   # Rules specified here may also be granted to the Insights OPA plugin,
-  # see also the insights-agent chart values for opa.
+  # see also the insights-al  qgent chart values for opa.
```
Comment (rbren) [inline stable/insights-admission/values.yaml]:
```suggestion
  # see also the insights-agent chart values for opa.
```

File: stable/insights-agent/templates/opa/rbac.yaml [hunk 4]
```diff
@@ -5,27 +5,71 @@ metadata:
...
   labels:
     app: insights-agent
-{{- with .Values.opa.additionalAccess }}
+{{- if .Values.opa.defaultTargetResources }}
 ---
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
-  name: {{ include "insights-agent.fullname" $ }}-opa-additional
+  name: {{ include "insights-agent.fullname" $ }}-opa-deftargetresources
```
Comment (rbren) [inline stable/insights-agent/templates/opa/rbac.yaml]:
```suggestion
  name: {{ include "insights-agent.fullname" $ }}-opa-defaulttargetresources
```

Comment (rbren) [reply stable/insights-agent/templates/opa/rbac.yaml]: `deft` confused me - maybe we can find another way to shorten? It's probably short enough though.

File: stable/insights-agent/values.yaml [hunk 5]
```diff
@@ -157,17 +157,20 @@ opa:
     resources:
     - pods
     - replicationcontrollers
-  # opa.admissionRulesAsTargetResources -- IF the admission controller is
-  # enabled, the APIGroups and Resources found in admission rules will be
-  # used as Kubernetes target resources for this plugin.
-  # This also grants RBAC rules to this plugin that have been specified in
-  # the insights-admission chart `serviceAccount.rbac.additionalAccess` value.
-  # See also: the insights-admission chart `webhookConfig.rules` value.
+  # opa.admissionRulesAsTargetResources -- IF the admission controller is enabled,
```
Comment (rbren) [inline stable/insights-agent/values.yaml]:
```suggestion
  # opa.admissionRulesAsTargetResources -- If the admission controller is enabled,
```

File: stable/insights-agent/values.yaml [hunk 6]
```diff
@@ -157,17 +157,20 @@ opa:
...
+  # will be added as OPA Kubernetes target resources, plus
+  # supporting RBAC rules are granted to OPA, if the insights-admission
+  # `webhookConfig.rulesAutoRBAC` value is also set.
   admissionRulesAsTargetResources: true
   # opa.targetResources -- A user-specified list of Kubernetes targets to which
   # OPA policies will be applied.
   # Each target requires a list of apiGroups and a list of Resources.
-  targetResources: {}
+  targetResources: []
+  # opa.targetResourcesAutoRBAC -- AUtomatically add RBAC rules allowing get
```
Comment (rbren) [inline stable/insights-agent/values.yaml]:
```suggestion
  # opa.targetResourcesAutoRBAC -- Automatically add RBAC rules allowing get
```

**Review by rbren — APPROVED** [review]
Some minor typos, but otherwise lgtm!

File: stable/insights-agent/templates/opa/rbac.yaml (hunk 4 above)
Comment (ivanfetch-wt) [reply stable/insights-agent/templates/opa/rbac.yaml]: I verified there isn't a 63 character name limit here (which we could be close to depending on the Helm release name), and changed these `deftargetresources` names to the full `defaulttargetresources` name.

**Pull request merged**